### PR TITLE
Add Rails-style YAML support to i18n-cli

### PR DESCRIPTION
**Pull request by berfarah**
In order to use this with Rails projects, we're adding Rails support to the CLI

### CODE REVIEW

**Comment by berfarah** [comment]
So technically this isn't a 50 line diff - but yarn is half of this PR. :)

File: i18n-1490118652166.csv [hunk 1]
```diff
@@ -0,0 +1 @@
+"bundle","key","en-US"
```
Comment (kylealwyn) [inline i18n-1490118652166.csv]: what's this guy doing in here?

**Comment by berfarah** [comment]
Actually, I hacked in the file extension into this PR. I should probably do more refactoring to properly allow for multiple locales with Rails.

File: src/commands/bundles.js [hunk 2]
```diff
@@ -21,9 +21,9 @@ const generateLocaleExports = (adapter, outputPath, locales) => {
   return writeFile(localeExportPath, i18nIndexFile);
 }
 
-const generateBundleExports = (adapter, outputPath, files, locales) => {
+const generateBundleExports = (adapter, outputPath, files = [], locales) => {
   const { index } = adapter;
-  if (!files || !files.length || files.every(f => !f.newFile)) return;
+  if (!files.length) return;
 
   const bundleList = (files || []).map((file) => (
```
Comment (kylealwyn) [inline src/commands/bundles.js]: can remove `(files || [])`

File: src/adapters/rails.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const { safeLoad, safeDump } = require('js-yaml');
+const fs = require('fs');
+
+module.exports = {
+  extension: '.en.yml',
```
Comment (kylealwyn) [inline src/adapters/rails.js]: should be `.lang.yml`, no?

File: test/adapters/rails.spec.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  describe('serialize()', () => {
+    it('returns a valid YAML export', () => (
+      expect(subject.serialize({ foo: 'bar' }, 'en', 'baz')).to.eql(sample)
+    ));
+  });
+
+  describe('read()', () => {
+    it('reads YAML into a JS object', () => (
+      subject.read(fixture).then((object) => (
+        expect(object).to.eql({ foo: 'bar' })
```
Comment (kylealwyn) [inline test/adapters/rails.spec.js:22]: why don't i get the full lang object back?

**Review by kylealwyn — CHANGES_REQUESTED** [review]

File: i18n-1490118652166.csv (hunk 1 above)
Comment (berfarah) [reply i18n-1490118652166.csv]: Oh whoops - thanks.

File: src/commands/bundles.js (hunk 2 above)
Comment (berfarah) [reply src/commands/bundles.js]: Yep!

File: src/adapters/rails.js (hunk 3 above)
Comment (berfarah) [reply src/adapters/rails.js]: For rails, no. Unless lang is dynamically replaced?

File: test/adapters/rails.spec.js (hunk 4 above)
Comment (berfarah) [reply test/adapters/rails.spec.js:22]: For spreadsheet upload/diffing we don't want the full lang object.

File: src/adapters/rails.js (hunk 3 above)
Comment (kylealwyn) [reply src/adapters/rails.js]: just read up on rails i18n. got it.

Comment (berfarah) [reply src/adapters/rails.js]: But I feel like it should be possible to replace this :\ I'm don't feel good about making the extension a function for everything though. ill think more about it.

**Review by kylealwyn — APPROVED** [review]
⭐️

**Comment by berfarah** [comment]
@kylealwyn Can you release this to our NPM? I don't currently know how. Or we can do it together tomorrow :)

**Comment by kylealwyn** [comment]
Yep! I'll show ya tomorrow

**Pull request merged**